kf_lib_data_ingest/validation/test_input_structure.py

```diff
@@ -0,0 +1,230 @@
+        constants.OUTCOME.DISEASE_RELATED.YES,
+        constants.OUTCOME.DISEASE_RELATED.NO,
+    },
+    CONCEPT.OUTCOME.VITAL_STATUS: {
+        constants.OUTCOME.VITAL_STATUS.ALIVE,
+        constants.OUTCOME.VITAL_STATUS.DEAD,
+    },
+    # PHENOTYPE
+    CONCEPT.PHENOTYPE.EVENT_AGE_DAYS: check_age,
+    CONCEPT.PHENOTYPE.OBSERVED: {
+        constants.PHENOTYPE.OBSERVED.YES,
+        constants.PHENOTYPE.OBSERVED.NO,
+    },
+    # READ_GROUP
+    CONCEPT.READ_GROUP.LANE_NUMBER: check_positive,
+    CONCEPT.READ_GROUP.QUALITY_SCALE: {
+        constants.READ_GROUP.QUALITY_SCALE.ILLUMINA13,
+        constants.READ_GROUP.QUALITY_SCALE.ILLUMINA15,
+        constants.READ_GROUP.QUALITY_SCALE.ILLUMINA18,
+        constants.READ_GROUP.QUALITY_SCALE.SANGER,
+        constants.READ_GROUP.QUALITY_SCALE.SOLEXA,
+    },
+    # READ_GROUP_GENOMIC_FILE
+    # SEQUENCING_CENTER
+    # SEQUENCING_EXPERIMENT
+    CONCEPT.SEQUENCING.DATE: check_date,
+    CONCEPT.SEQUENCING.STRATEGY: {
+        constants.SEQUENCING.STRATEGY.TARGETED,
+        constants.SEQUENCING.STRATEGY.WXS,
+        constants.SEQUENCING.STRATEGY.WGS,
+        constants.SEQUENCING.STRATEGY.RNA,
+        constants.SEQUENCING.STRATEGY.MRNA,
+        constants.SEQUENCING.STRATEGY.LINKED_WGS,
+        constants.COMMON.OTHER,
+    },
+    # TODO: CONCEPT.SEQUENCING.LIBRARY_PREP - there is no concept for this
+    # TODO: CONCEPT.SEQUENCING.LIBRARY_SELECTION - no concept for this either
+    CONCEPT.SEQUENCING.LIBRARY_STRAND: {
+        constants.SEQUENCING.STRAND.UNSTRANDED,
+        constants.SEQUENCING.STRAND.FIRST,
+        constants.SEQUENCING.STRAND.SECOND,
+        constants.COMMON.OTHER,
+    },
+    CONCEPT.SEQUENCING.MAX_INSERT_SIZE: check_positive,
+    CONCEPT.SEQUENCING.MEAN_DEPTH: check_positive,
+    CONCEPT.SEQUENCING.MEAN_INSERT_SIZE: check_positive,
+    CONCEPT.SEQUENCING.MEAN_READ_LENGTH: check_positive,
+    CONCEPT.SEQUENCING.PLATFORM: {
+        constants.SEQUENCING.PLATFORM.ION_TORRENT,
+        constants.SEQUENCING.PLATFORM.SOLID,
+        constants.SEQUENCING.PLATFORM.ILLUMINA,
+        constants.SEQUENCING.PLATFORM.LS454,
+        constants.SEQUENCING.PLATFORM.GENOMICS,
+        constants.SEQUENCING.PLATFORM.PACBIO,
+        constants.COMMON.OTHER,
+    },
+    CONCEPT.SEQUENCING.TOTAL_READS: check_positive,
+    # SEQUENCING_EXPERIMENT_GENOMIC_FILE
+    # STUDY
+    CONCEPT.STUDY.RELEASE_STATUS: {
+        constants.STUDY.STATUS.FAILED,
+        constants.STUDY.STATUS.PENDING,
+        constants.STUDY.STATUS.PUBLISHING,
+        constants.STUDY.STATUS.CANCELED,
+        constants.STUDY.STATUS.STAGED,
+        constants.STUDY.STATUS.RUNNING,
+        constants.STUDY.STATUS.PUBLISHED,
+        constants.STUDY.STATUS.WAITING,
+    },
+    # STUDY_FILE
+    CONCEPT.STUDY_FILE.AVAILABILITY: {
+        constants.GENOMIC_FILE.AVAILABILITY.IMMEDIATE,
+        constants.GENOMIC_FILE.AVAILABILITY.COLD_STORAGE,
+    },
+    # TASK
+    # TASK_GENOMIC_FILE
+}
```

Comment:
Maybe consider adding a validation function for enumerations so that every value in the `INPUT_VALIDATION` dict will have a standard calling interface. Right now the user of `INPUT_VALIDATION` needs to know whether the key has a set or function as its value: 

```
def check_enum_wrap(enum):
    def check_enum(x):
        return x in (enum | NULL_VALUES)
    return check_enum

INPUT_VALIDATION = {
    k: (check_enum_wrap(v) if isinstance(v, set) else v)
    for k, v in INPUT_VALIDATION.items()
}
```